src/App.tsx

```diff
@@ -1,11 +1,32 @@
-import React from 'react';
+import React, {useState} from 'react';
 import {HomePage} from "./HomePage/HomePage";
 import {ErrorBoundary} from "./HomePage/components/ErrorBoundary/ErrorBoundary";
+import {mockedMovies, Movies} from "./Mocks";
 
+export interface AppInterface {
+    movies: Movies,
+    app: {
+        title: string,
+        logo: string
+    }
+}
+
+const CategoriesContext = React.createContext(['All', 'Documentary', 'Comedy', 'Horror', 'Crime']);
+export const CategoriesProvider = CategoriesContext.Provider
+export default CategoriesContext
 
 export function App() {
+    const [appState] = useState({
+        movies: mockedMovies,
```

Comment:
I would recommend to leave only this dat in state, declare setter for it. Then hold both movies and setter in context -> this give you ability to conveniently access movies list and change it from any child component instead of passing them as props